frontend/src/utilities/__tests__/imageUtils.spec.ts

```diff
@@ -0,0 +1,421 @@
+    const buildStatuses: BuildStatus[] = [
+      {
+        name: 'Build1',
+        imageTag: 'image:tag1',
+        status: BUILD_PHASE.complete,
+        timestamp: '2022-01-01',
+      },
+    ];
+
+    const image = {
+      name: 'image',
+      tags: [{ name: 'tag1' }] as ImageTagInfo[],
+    } as ImageInfo;
+    expect(getDefaultTag(buildStatuses, image)).toEqual({ name: 'tag1' });
+  });
+});
+describe('getTagForImage', () => {
+  it('should return the selected tag if image has multiple tags and a selected tag is provided', () => {
+    const buildStatuses: BuildStatus[] = [
+      {
+        name: 'Build1',
+        imageTag: 'imageName:tag1',
+        status: BUILD_PHASE.complete,
+        timestamp: '2022-01-01',
+      },
+    ];
+
+    const image = {
+      name: 'imageName',
+      tags: [{ name: 'tag1' }, { name: 'tag2' }] as ImageTagInfo[],
+    } as ImageInfo;
+
+    const selectedImage = 'imageName';
+    const selectedTag = 'tag2';
+
+    const result = getTagForImage(buildStatuses, image, selectedImage, selectedTag);
+
+    expect(result).toEqual({ name: 'tag2' });
+  });
+  it('should return the default tag if image has multiple tags and no selected tag is provided', () => {
+    const buildStatuses: BuildStatus[] = [
+      {
+        name: 'Build1',
+        imageTag: 'imageName:tag1',
+        status: BUILD_PHASE.complete,
+        timestamp: '2022-01-01',
+      },
+    ];
+
+    const image = {
+      default: true,
+      name: 'imageName',
+      tags: [
+        { name: 'tag1', default: true },
+        { name: 'tag2', default: false },
+      ] as ImageTagInfo[],
+    } as ImageInfo;
+
+    const result = getTagForImage(buildStatuses, image);
+
+    expect(result).toEqual({ name: 'tag1', default: true });
+  });
+
+  it('should return undefined if image has no tags', () => {
+    const buildStatuses: BuildStatus[] = [
+      {
+        name: 'Build1',
+        imageTag: 'imageName:tag1',
+        status: BUILD_PHASE.complete,
+        timestamp: '2022-01-01',
+      },
+    ];
+
+    const image = {
+      name: 'imageName',
+      tags: undefined as unknown as ImageTagInfo[],
+    } as ImageInfo;
+
+    const result = getTagForImage(buildStatuses, image);
+
+    expect(result).toBeUndefined();
+  });
+});
+describe('getImageTagVersion', () => {
+  it('should return the selected tag version if image has multiple tags and a selected tag is provided', () => {
+    const buildStatuses: BuildStatus[] = [
+      {
+        name: 'Build1',
+        imageTag: 'imageName:tag1',
+        status: BUILD_PHASE.complete,
+        timestamp: '2022-01-01',
+      },
+    ];
+
+    const image = {
+      name: 'imageName',
+      tags: [{ name: 'tag1' }, { name: 'tag2' }] as ImageTagInfo[],
+    } as ImageInfo;
+
+    const selectedImage = 'imageName';
+    const selectedTag = 'tag2';
+
+    const result = getImageTagVersion(buildStatuses, image, selectedImage, selectedTag);
+
+    expect(result).toEqual('tag2');
```

Comment:
nit (here and many other places in this file):
use `toBe` when comparing strings.